### DISCUSSION TITLE
How to set default cmake build system

### DISCUSSION THREAD

**Discussion by anderslanglands** [AcademySoftwareFoundation/rez#1333] [Q&A] [answered]
Hello, rez currently defaults to using nmake on Windows, which is awful. Is there a way to set it to ninja by default instead? I tried adding `"cmake_build_system": "ninja"` to my .reconfig but no dice. I also tried following through the code to see where it's set but can't find it other than the args parser.

EDIT: I also tried:
```python
{
    "cmake": {
        "build_system": "ninja"
    }
}
```

**Comment by maxnbk** [comment, marked as answer]
This probably seems counterintuitive, but try:
`plugins = {'build_system': {'cmake': {'build_system': 'ninja'}}}`

If I'm not wrong:
rez has a default build system (`cmake`), but `cmake` has a default build system (`nmake` in this case). I assume what you want is to change cmake's default `build_system`, not rez default `build_system`, unless ninja's scope has expanded drastically since I last checked.

Keep in mind that this correlates to the difference between `rez-build`'s `-b` flag, and `--bt` + `--bs` flags. You'll see that when cmake is your default build system, the `--bt` + `--bs` flags are cmake-specific flags.

**Reply by anderslanglands** [reply]
Thanks! That did the trick. My .rezconfig is now:
```python
{
    "default_shell": "pwsh",
    "plugins": {
        "build_system": {
            "cmake": {
                "build_system": "ninja"
            }
        }
    }
}
```
